src/test/java/com/microsoft/sqlserver/jdbc/ActivityIDTest.java

```diff
@@ -0,0 +1,171 @@
+                        ResultSet rs = s.executeQuery("SELECT @@VERSION AS 'SQL Server Version'")) {
+                    while (rs.next()) {
+                        rs.getString(1);
+                    }
+                } catch (SQLException e) {
+                    fail(e.toString());
+                }
+                latch.countDown();
+            });
+        });
+        latch.await();
+        es.shutdown();
+        assertEquals(0, ActivityCorrelator.getActivityIdTlsMap().size());
+    }
+    
+    @Test
+    public void testActivityIDPooled() throws Exception {
+        int poolsize = 10;
+        int numPooledExecution = 200;
+        
+        HikariConfig config = new HikariConfig();
+        config.setJdbcUrl(connectionString);
+        config.setMaximumPoolSize(poolsize);
+        ExecutorService es = Executors.newFixedThreadPool(poolsize);
+        CountDownLatch latchPoolOuterThread = new CountDownLatch(1);
+        Thread t = new Thread(new Runnable() {
+            CountDownLatch latchPool = new CountDownLatch(numPooledExecution);
+            public void run() {
+                HikariDataSource ds = new HikariDataSource(config);
+                    es.execute(() -> {
+                        IntStream.range(0, numPooledExecution).forEach(i -> {
+                            try (Connection c = ds.getConnection(); Statement s = c.createStatement();
+                                    ResultSet rs = s.executeQuery("SELECT @@VERSION AS 'SQL Server Version'")) {
+                                while (rs.next()) {
+                                    rs.getString(1);
+                                }
+                            } catch (SQLException e) {
+                                fail(e.toString());
+                            }
+                            latchPool.countDown();
+                        });
+                    });
+                try {
+                    latchPool.await();
+                } catch (InterruptedException e) {
+                    fail(e.toString());
+                } finally {
+                    if (null != ds) {
+                        es.shutdown();
+                        ds.close();
+                    }
+                }
+                latchPoolOuterThread.countDown();
+            }
+        });
+        t.run();
+        latchPoolOuterThread.await();
+        
+        try {
+            try (Connection con = getConnection(); Statement stmt = con.createStatement()) {
+                stmt.execute("SELECT @@VERSION AS 'SQL Server Version'");
+            }
+        } catch (SQLException e) {
+            fail(e.toString());
+        }
+        
+        try {
+            assertEquals(0, ActivityCorrelator.getActivityIdTlsMap().size());
+        } catch (ComparisonFailure e) {
+            Set<Thread> threadSet = Thread.getAllStackTraces().keySet();
+            logger.fine("List of threads alive:");

```

Comment:
Logger is disabled when JUnit tests are running, writing log messages doesn't help if a failure occurs randomly. Use System.out.println() to print this information that needs to be captured immediately.